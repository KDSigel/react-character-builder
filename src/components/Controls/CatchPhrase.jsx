import React from 'react'

export default function CatchPhrase({newCatchphrase, setNewCatchphrase, handleClick}) {
    return (
        <div>
            <label>
            <p>Add a catch phrase:
                <input
                type='text'
                value={newCatchphrase}
                onChange={(e) => setNewCatchphrase(e.target.value)} />
                <button type='button' onClick={handleClick}>Add</button>
            </p>
            </label>
        </div>
    )
}
