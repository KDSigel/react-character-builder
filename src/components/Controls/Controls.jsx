import React from 'react'

export default function Controls({
    head, 
    setHead,
    torso,
    setTorso,
    legs,
    setLegs,
}) {

const headOptions = ['Fry', 'MemeMan', 'ShibaInu']
const torsoOptions = ['Aladdin', 'Homer', 'Spidey']
const legsOptions = ['SkateObama', 'SpongeBob', 'SpiderLegs']

    return (
        <div>
            <label>
                <p>Head 
                    <select onChange={(e) => setHead(e.target.value)} value={head}>
                        {headOptions.map((headOption) => (
                            <option value={headOption} key={headOption}>{headOption}</option>
                        ))
                        }
                    </select>
                    </p>
            </label>
            <label>
                <p>Torso 
                    <select onChange={(e) => setTorso(e.target.value)} value={torso}>
                        {torsoOptions.map((torsoOption) => (
                            <option key={torsoOption}>{torsoOption}</option>
                        ))
                        }
                    </select>
                    </p>
            </label>
            <label>
                <p>Legs 
                    <select onChange={(e) => setLegs(e.target.value)} value={legs}>
                        {legsOptions.map((legsOption) => (
                            <option key={legsOption}>{legsOption}</option>
                        ))
                        }
                    </select>
                    </p>
            </label>
        </div>
    )
}
