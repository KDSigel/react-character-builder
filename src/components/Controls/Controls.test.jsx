
import {render} from '@testing-library/react';
import Controls from './Controls';

it('should render our Picker component', () => {

  const {container} = render(<Controls />);
  expect(container).toMatchSnapshot();

});