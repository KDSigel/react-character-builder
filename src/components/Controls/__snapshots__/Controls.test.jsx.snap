// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our Picker component 1`] = `
<div>
  <div>
    <label>
      <p>
        Head
        <select>
          <option
            value="Fry"
          >
            Fry
          </option>
          <option
            value="MemeMan"
          >
            MemeMan
          </option>
          <option
            value="ShibaInu"
          >
            ShibaInu
          </option>
        </select>
      </p>
    </label>
    <label>
      <p>
        Torso
        <select>
          <option>
            Aladdin
          </option>
          <option>
            Homer
          </option>
          <option>
            Spidey
          </option>
        </select>
      </p>
    </label>
    <label>
      <p>
        Legs
        <select>
          <option>
            SkateObama
          </option>
          <option>
            SpongeBob
          </option>
          <option>
            SpiderLegs
          </option>
        </select>
      </p>
    </label>
  </div>
</div>
`;
